### PR TITLE
feat(me): add support for database schemas (namespaces) outside of the Prisma schema

### PR DESCRIPTION
**Pull request by jkomyno**
This required moving `namespaces::Namespaces` from `migration-connector` to `introspection-connector`, to avoid circular dependencies. The introspection-engine doesn't yet support externally-provided namespaces.
As discussed with @pimeys, we might move `namespaces.rs` to `psl` in the future.

Contributes to https://github.com/prisma/prisma/issues/16455 and the PR https://github.com/prisma/prisma/pull/17034.

### CODE REVIEW

File: migration-engine/core/src/state.rs [hunk 1]
```diff
@@ -44,22 +46,31 @@ type ErasedConnectorRequest = Box<
...
+        } else {
+            self.initial_datamodel
+                .as_ref()
+                .and_then(|schema| schema.configuration.datasources.first())
+                .and_then(|ds| {
+                    let mut names = ds.namespaces.iter().map(|(ns, _)| ns.to_owned()).collect();
+                    Namespaces::from_vec(&mut names)
+                })
+        }
     }
```
Comment (jkomyno) [inline migration-engine/core/src/state.rs:74]: Either return the namespaces with which the `EngineState` was initialized, or take them from the schema

**Comment by jkomyno** [comment]
Integrated https://github.com/prisma/prisma-engines/pull/3560 in https://github.com/prisma/prisma-engines/pull/3553/commits/fe3162df40c4200e00e7984a22d8517de52b0fbd.

File: migration-engine/cli/tests/cli_tests.rs [hunk 2]
```diff
@@ -506,6 +506,221 @@ fn introspect_postgres(api: TestApi) {
...
+    .unwrap();
+    stdin.write_all(msg.as_bytes()).unwrap();
+    stdin.write_all(b"\n").unwrap();
+
+    let mut response = String::new();
+    stdout.read_line(&mut response).unwrap();
+
+    let expected = expect![[r#"
+        {"jsonrpc":"2.0","error":{"code":4466,"message":"An error happened. Check the data field for details.","data":{"is_panic":false,"message":"The introspected database was empty.","meta":null,"error_code":"P4001"}},"id":1}
+    "#]];
```
Comment (jkomyno) [inline migration-engine/cli/tests/cli_tests.rs:717]: This is of course wrong. I have added temporary `dbg!` statements in `state.rs` with no luck so far. Tests are run with `-- --nocapture`

**Comment by Jolg42** [comment]
I don't think this is needed anymore, I think the work was done in https://github.com/prisma/prisma-engines/pull/3574/files

**Pull request closed without merging**